src/BinaryObject.h

```diff
@@ -26,7 +27,9 @@ class BinaryObject {
   void setCpuSubType(CpuType type);
 
   [[nodiscard]] bool isLittleEndian() const;
-  void setLittleEndian(bool little);
+  [[nodiscard]] bool isBigEndian() const;
+
+  void setEndianness(Constants::Endianness endianness);
```

Comment:
And changing into having only the two set+get endianness functions here, like in Reader, too.